src/main.rs

```diff
@@ -158,12 +158,11 @@ async fn main() -> Result<()> {
     let credentials_provider = DefaultCredentialsProvider::new().unwrap();
     let s3 = S3Client::new_with(http_client, credentials_provider, Region::default());
 
-    setup_cancel_handler();
-
     use Command::*;
 
     match cli.cmd {
         Put { bucket, key, file } => {
+            setup_upload_ctrlc_handler();
```

Comment:
I'm just curious what the `ctrlc` stands for/means.  Controller?

Reply:
Nevermind, I found this https://docs.rs/ctrlc/3.1.7/ctrlc/

Reply:
fixed